### PR TITLE
don't zero out the attention_mask when using sliding window with flash attention

### PR DESCRIPTION
**Pull request by winglian**
# What does this PR do?

Flash attention has it's own sliding window argument,  but currently the attention_mask is clobbered with sliding window leading to incorrect/large loss values (and also breaks sample packing in axolotl):

Here's the existing loss with flash attention and no sample packing (LoRA):

Screenshot 2024-06-27 at 12 17 12 PM

Here's the loss with the fix (no sample packing):

Screenshot 2024-06-27 at 12 20 09 PM

Fixes # (issue)

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#create-a-pull-request),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

@muellerzr @SunMarc  @ArthurZucker

### CODE REVIEW

**Review by ArthurZucker — APPROVED** [review]
We can indeed skip that step for FA2! as the sliding argument is enough

**Comment by ArthurZucker** [comment]
Checkout https://pypi.org/project/transformers/4.42.2/ which includes this commit!
